pkg/virt-controller/watch/migration_test.go

```diff
@@ -1250,7 +1256,11 @@ var _ = Describe("Migration watcher", func() {
 			if phase == virtv1.MigrationFailed {
 				// This finalizer is added by the mutation webhook during creation
 				migration.Finalizers = append(migration.Finalizers, virtv1.VirtualMachineInstanceMigrationFinalizer)
-				shouldExpectMigrationFinalizerRemoval(migration)
+				if initializeMigrationState {
+					shouldExpectMigrationStateUpdated(migration, vmi.Status.MigrationState)
+				} else {
+					shouldExpectMigrationFinalizerRemoval(migration)
+				}
```

Comment:
Can you explain why the previous `shouldExpectMigrationFinalizerRemoval` function is in the else statement, please?
Previously that was checked in every `Entry`, while with this change only if `initializedMigrationState` is false.
Thanks